rest/perf_routes.go

```diff
@@ -161,6 +161,60 @@ func (h *perfGetByTaskIdHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(perfResults)
 }
 
+///////////////////////////////////////////////////////////////////////////////
+//
+// GET /perf/task_id/{task_id}/exists
+
+type perfExistsByTaskIdHandler struct {
+	opts data.PerformanceOptions
+	sc   data.Connector
+}
+
+func makeExistsPerfByTaskId(sc data.Connector) gimlet.RouteHandler {
+	return &perfExistsByTaskIdHandler{
+		sc: sc,
+	}
+}
+
+// Factory returns a pointer to a new perfGetByTaskIdHandler.
+func (h *perfExistsByTaskIdHandler) Factory() gimlet.RouteHandler {
+	return &perfExistsByTaskIdHandler{
+		sc: h.sc,
+	}
+}
+
+// Parse fetches the task_id from the http request.
+func (h *perfExistsByTaskIdHandler) Parse(_ context.Context, r *http.Request) error {
+	h.opts.TaskID = gimlet.GetVars(r)["task_id"]
+	h.opts.Tags = r.URL.Query()[tags]
+	return nil
+}
+
+// Run calls the data FindPerformanceResults function and returns the
+// PerformanceResults from the provider.
+func (h *perfExistsByTaskIdHandler) Run(ctx context.Context) gimlet.Responder {
+	perfResults, err := h.sc.FindPerformanceResults(ctx, h.opts)
+	if err != nil {
+		err = errors.Wrapf(err, "problem getting performance results by task id '%s'", h.opts.TaskID)
+		grip.Error(message.WrapError(err, message.Fields{
+			"request": gimlet.GetRequestID(ctx),
+			"method":  "GET",
+			"route":   "/perf/task_id/{task_id}/exists",
+			"task_id": h.opts.TaskID,
+		}))
+		return gimlet.MakeJSONErrorResponder(err)
+	}
+	if len(perfResults) > 0 {
```

Comment:
nit: normally, early return paths are for the unhappy case (i.e. perf results don't exist) and the "main code path" (i.e. the one without the if statements) is for the happy path. Can you invert L207-215 so that it reads more like this?
```
if len(perfResults) == 0 {
    // Return because we've hit the unhappy path.
    return <error response>
}

// Return the real result in the happy path.
return gimlet.NewJSONResponse(result)
```